src/s2/s2polyline.h

```diff
@@ -397,6 +397,8 @@ class S2Polyline final : public S2Region {
     explicit OwningShape(std::unique_ptr<const S2Polyline> polyline)
         : Shape(polyline.get()), owned_polyline_(std::move(polyline)) {}
 
+    ~OwningShape() override;
```

Comment:
Can you add a comment like
    // `S2Polyline` is an incomplete type here.  Due to `unique_ptr`,
    // `~OwningShape` must be defined after `S2Polyline` is complete, so
    // it cannot be implicit.
?

Reply:
Makes sense, done.